extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/DefaultTenantConfigResolver.java

```diff
@@ -189,12 +195,9 @@ boolean isSecurityEventObserved() {
         return securityEventObserved;
     }
 
-    void setSecurityEventObserved(boolean securityEventObserved) {
-        this.securityEventObserved = securityEventObserved;
-    }
-
-    Event<SecurityEvent> getSecurityEvent() {
-        return securityEvent;
+    void fireSecurityEvent(SecurityEvent event) {
+        securityEvent.fire(event);
```

Comment:
@michalvavrik I'm not sure about `SecurityEventHelper` refactoring like `fireSuccessEvents`, `fireFailureEvents`, they can't take care of this code. Can we just make `SecureEventHelper.fire(securityEvent, event)` only used everywhere ? I thought we agreed to have a simple method added to it and now it looks totally different. Can you think of how  `SecureEventHelper` can support this code too ?

Reply:
IMHO it is worth it, not a big deal, but with new event types to be added in time, it is worth having a simple option to encapsulate `fire`, `fireAsync` for all events. This is the very last thing I'd like to ask for, I've been already prepared to approve, it is very close IMHO

Reply:
> I thought we agreed to have a simple method added to it and now it looks totally different.

That was necessary, previously it was a record - **data carrier** - with only methods creating the record, now that you wanted to add utility method, it had to look differently in order to not disuse records.

> Can we just make SecureEventHelper.fire(securityEvent, event) only used everywhere ?

I thought how I refactored `SecurityEventHelper` allowed me to limit repeating of same pattern with booleans and `Event`. I'm not sure method like this

```
public static <T extends SecurityEvent> void fire(Event<T> securityEvent, T event) {
     securityEvent.fire(event);
    securityEvent.fireAsync(event);
}
```
is super useful, I think what I did limited repetition better. I can add this method to `SecurityEventHelper` @sberyozkin , but I don't see a reason to reduce helper as it looks now.

Reply:
Done.